File: Practica1/Backend-python/app/repositories/notification_repository.py
from app.config import get_connection

def get_notifications_by_user(id_usuario):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("""
        SELECT id_notificacion, mensaje, leida, fecha
        FROM notificaciones
        WHERE id_usuario = %s
        ORDER BY fecha DESC
    """, (id_usuario,))

    notifications = cursor.fetchall()

    cursor.close()
    conn.close()

    return notifications


def create_notification(id_usuario, mensaje):
    conn = get_connection()
    cursor = conn.cursor()

    cursor.execute("""
        INSERT INTO notificaciones (id_usuario, mensaje)
        VALUES (%s, %s)
    """, (id_usuario, mensaje))

    conn.commit()

    cursor.close()
    conn.close()

File: Practica1/Backend-Nodejs/src/repositories/notification.repository.js
const pool = require("../config/db");

async function createNotification(id_usuario, mensaje) {
  await pool.query(
    `
    INSERT INTO notificaciones (id_usuario, mensaje)
    VALUES ($1, $2)
    `,
    [id_usuario, mensaje]
  );
}

async function getNotificationsByUser(id_usuario) {
  const result = await pool.query(
    `
    SELECT id_notificacion, mensaje, leida, fecha
    FROM notificaciones
    WHERE id_usuario = $1
    ORDER BY fecha DESC
    `,
    [id_usuario]
  );
  return result.rows;
}

module.exports = {
  createNotification,
  getNotificationsByUser
};